lib/config.js

```diff
@@ -186,17 +188,21 @@ class Config {
     /**
      * Config options
      * @param {Object} options Options to be passed in
+     * @param {Linter} linterContext Linter instance object
      */
-    constructor(options) {
+    constructor(options, linterContext) {
```

Comment:
Thats why I didnt not touch any core logic pieces. I think this a step towrads that and we can slowly improve that in the future.